src/js/components/covid19/AmountsVisualization.jsx

```diff
@@ -303,16 +304,14 @@ const AmountsVisualization = ({
     useLayoutEffect(() => {
         const { text: textInfo } = rectangleMapping._remainingBalance;
         const questionRef = _remainingBalanceQuestion.current?.getBoundingClientRect();
-        if (remainingBalanceLineData) {
-            setRemainingBalanceQuestionData({
-                y: remainingBalanceLineData.y2 - remainingBalanceValueData?.height - spacingBetweenLineAndText,
-                x: remainingBalanceLineData.x1 - ((questionRef?.width || 0) + spacingBetweenLineAndText),
-                height: questionRef?.height || 0,
-                text: textInfo.question,
-                className: `amounts-text__question ${!questionRef ? 'white' : ''}`
-            });
-        }
-    }, [remainingBalanceLineData]);
+        setRemainingBalanceQuestionData({
+            y: (remainingBalanceLineData?.y2 || 0) - (remainingBalanceValueData?.height || 0) - spacingBetweenLineAndText,
+            x: (remainingBalanceLineData?.x1 || 0) - ((questionRef?.width || 0) + spacingBetweenLineAndText),
+            height: questionRef?.height || 0,
+            text: textInfo.question,
+            className: `amounts-text__question ${!questionRef ? 'white' : ''}`
+        });
+    }, [remainingBalanceLineData, obligationValueData]);
```

Comment:
Should this dependency be `remainingBalanceValueData` instead of (or in addition to) `obligationValueData`?

Reply:
Yeah, I mainly tried to get these hooks to use update themselves logically but if I add remainingBalanceValueData it will launch an infinite loop.